enumerables.rb

```diff
@@ -0,0 +1,269 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    arr = is_a?(Array) ? self : to_a
+    is_a?(Hash) ? arr.length.times { |i| yield(arr[i][0], arr[i][1]) } : arr.length.times { |i| yield(arr[i]) }
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    arr = to_a
+    (0...arr.length).my_each do |i|
+      yield(arr[i], i)
+    end
+    self
+  end
+
+  def my_map(&proc)
+    my_arr = []
+    hash = {}
+    if block_given?
+      is_a?(Hash) ? my_each { |k, v| hash[k] = yield(k, v) } : my_each { |x| my_arr << yield(x) }
+    else
+      my_each { |x| proc.call(x) }
+    end
+    is_a?(Hash) ? hash : my_arr
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    if is_a?(Array)
+      my_arr = []
+      my_each do |i|
+        my_arr << i if yield(i)
+      end
+      my_arr
+    elsif is_a?(Hash)
+      my_hash = {}
+      my_each do |i, j|
+        my_hash.store(i, j)
+      end
+    end
+  end
+
+  def my_all?(*args)
+    is_true = true
+    if block_given?
+      if is_a?(Hash)
+        to_a.my_each do |k, v|
+          is_true = false unless yield(k, v)
+        end
+      else my_each { |x| is_true = false unless yield(x) }
+      end
+    elsif args[0].is_a?(Integer)
+      my_each { |x| is_true = false unless args[0] == x }
+    elsif args[0].nil?
+      my_each { |x| is_true = false if x == false || x.nil? }
+    elsif args[0].is_a?(Regexp)
+      my_each { |x| is_true = false if x.match(args[0]).nil? }
+    else
+      my_each { |x| is_true = false unless x.is_a?(args[0]) }
+    end
+    is_true
+  end
+
+  def my_count(num = 0)
+    sum = 0
+    length.times { |x| sum += 1 if self[x] == num } if num != 0
+    sum = max if !block_given? && num.zero?
```

Comment:
You should return length according to the definition of #my_count in this documentation: https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png
when no block or arg given.